### DISCUSSION TITLE
Cron jobs on Docker Swarm

### DISCUSSION THREAD

**Discussion by CarlosEspejo** [BretFisher/ama#182] [Q&A]
Is it possible to run a service that gets created on a daily schedule, that will run for a minute or two then exit but without Docker Swarm thinking that the container failed?

**Comment by BretFisher** [comment]
This isn't a built-in feature yet, however, check:

https://github.com/alexellis/jaas

**Comment by CarlosEspejo** [comment]
I had to deploy something this week so I went with a single VM with docker daemon, the docker run commands in shell scripts and CRON schedules to trigger them. But will give this a go when I build my docker swarm.

Thanks 😄

**Comment by BretFisher** [comment]
More on this topic. Not everything needs a fancy multi-server cron-like engine with retry, queues, etc.  

[Here's an example swarm service](https://github.com/BretFisher/dogvscat/blob/master/stack-prune.yml) that will run on every node and once a day, clean up docker images buy running prune.

[Here's another example](https://github.com/BretFisher/dogvscat/blob/master/stack-rexray.yml), this one is run once on each node, to install a docker plugin for volume drivers.

**Comment by gerassimos** [comment]
Hello 
I was looking also for something similar to perform periodic DB backup of a postgres DB container. I found [this](https://hub.docker.com/r/prodrigestivill/postgres-backup-local) docker image that seams to be a good solution for this specific case. I do not know if there are better ways to perform the periodic backups of DB containers
